src/index.ts

```diff
@@ -156,6 +158,24 @@ export default class Matic extends ContractsBase {
     return this.depositManager.depositEther(amount, options)
   }
 
+  async depositDataByHash(txHash: string) {
+    const depositReceipt = await this.web3Client.parentWeb3.eth.getTransactionReceipt(txHash)
+    if(!depositReceipt) {
+      return 'Transaction hash is not Found'
+    }
+    const newDepositEvent = depositReceipt.logs.find(
+      l => l.topics[0].toLowerCase() === DepositManager.NEW_DEPOSIT_EVENT_SIG,
+    )
+
+    const data = newDepositEvent.data
+    const depositId = parseInt(data.substring(data.length - 64), 16)
```

Comment:
- Use bigNumber (bn.js). 256 bit integers will overflow when using `parseInt`.
- Isn't the data 0x prefixed? - in that case you'll need to slice 66 characters.

Reply:
The prefix doesn't matter, It will pick last 64 characters
parsing will return the same data